clients/include/cblas_interface.hpp

```diff
@@ -1625,6 +1625,118 @@ void cblas_syr(
 }
 */
 
+// syr2
+template <typename T>
+inline void cblas_syr2(rocblas_fill uplo,
+                       rocblas_int  n,
+                       T            alpha,
+                       T*           x,
+                       rocblas_int  incx,
+                       T*           y,
+                       rocblas_int  incy,
+                       T*           A,
+                       rocblas_int  lda);
+
+template <>
+inline void cblas_syr2(rocblas_fill uplo,
+                       rocblas_int  n,
+                       float        alpha,
+                       float*       x,
+                       rocblas_int  incx,
+                       float*       y,
+                       rocblas_int  incy,
+                       float*       A,
+                       rocblas_int  lda)
+{
+    cblas_ssyr2(CblasColMajor, CBLAS_UPLO(uplo), n, alpha, x, incx, y, incy, A, lda);
+}
+
+template <>
+inline void cblas_syr2(rocblas_fill uplo,
+                       rocblas_int  n,
+                       double       alpha,
+                       double*      x,
+                       rocblas_int  incx,
+                       double*      y,
+                       rocblas_int  incy,
+                       double*      A,
+                       rocblas_int  lda)
+{
+    cblas_dsyr2(CblasColMajor, CBLAS_UPLO(uplo), n, alpha, x, incx, y, incy, A, lda);
+}
+
+// No complex implementation of syr2, make a local version.
+template <typename T>
+inline void cblas_syr2_local(rocblas_fill uplo,
+                             rocblas_int  n,
+                             T            alpha,
+                             T*           xa,
+                             rocblas_int  incx,
+                             T*           ya,
+                             rocblas_int  incy,
+                             T*           A,
+                             rocblas_int  lda)
+{
+    if(n <= 0)
+        return;
+
+    T* x = (incx < 0) ? xa - ptrdiff_t(incx) * (n - 1) : xa;
+    T* y = (incy < 0) ? ya - ptrdiff_t(incy) * (n - 1) : ya;
+
+    if(uplo == rocblas_fill_upper)
+    {
+        for(int j = 0; j < n; ++j)
+        {
+            T tmpx = alpha * x[j * incx];
+            T tmpy = alpha * y[j * incy];
+            for(int i = 0; i <= j; ++i)
+            {
+                A[i + j * lda] = A[i + j * lda] + x[i * incx] * tmpy + y[i * incy] * tmpx;
+            }
+        }
+    }
+    else
+    {
+        for(int j = 0; j < n; ++j)
+        {
+            T tmpx = alpha * x[j * incx];
+            T tmpy = alpha * y[j * incy];
+            for(int i = j; i < n; ++i)
+            {
+                A[i + j * lda] = A[i + j * lda] + x[i * incx] * tmpy + y[i * incy] * tmpx;
+            }
+        }
+    }
+}
+
```

Comment:
It looks like blis has the symbols, you just have to hack in the extern "C" but what you did looks good too.
0000000000001ab0 T bli_csyr2
00000000000019f0 T bli_dsyr2
0000000000001930 T bli_ssyr2
0000000000001b80 T bli_zsyr2